test/modules/utils.test.js

```diff
@@ -78,6 +78,31 @@ test('createText() method',t => {
 
 		const actual = createText(str, embeds);
 		const expected = 'This is embed.js bar john foo';
-
-		t.same(actual, expected)
+		t.same(actual, expected);
+	
+		const str2 = 'Hello world';
+		const embeds2 = [ { index: 13, text: 'me' }, { index: 4, text: 'this' }, { index: 9, text: 'is'} ];
+		const actual2 = createText(str2, embeds2);
+		const expected2= 'Hello world this is me';
+		t.same(actual2, expected2);
+	
+		const str3 = 'Hello world';
+		const embeds3 = [];
+		const actual3 = createText(str3, embeds3);
+		const expected3= 'Hello world';
+		t.same(actual3, expected3);
+	
+		const str4 = '';
+		const embeds4 = [ { index: 13, text: 'me' }, { index: 4, text: 'this' }, { index: 9, text: 'is'} ];
+		const actual4 = createText(str4, embeds4);
+		const expected4= ' this is me';
```

Comment:
This shouldn't be the case. If the 1st string is empty then don't include space at beginning. Should be fixed in the function.